### PR TITLE
Online layer management (reorder and remove)

### PR DESCRIPTION
**Pull request by Piokor**

### CODE REVIEW

File: Backend.Core/ItemsManagers/LayersManager.cs [hunk 1]
```diff
@@ -2,33 +2,65 @@
...
 using System.Collections.Generic;
 using System.Linq;
+using Serilog;
 namespace Backend.Core
 {
     internal class LayersManager : IRoomItemsManager
     {
-        private readonly Dictionary<string, Layer> _layers = new Dictionary<string, Layer>();
+        //private readonly Dictionary<string, Layer> _layers = new Dictionary<string, Layer>();
+        private List<Layer> _layers = new List<Layer>();
```
Comment (mikasso) [inline Backend.Core/ItemsManagers/LayersManager.cs:11]: Not sure, if it is okay. If you want to sort them maybe just add index field to Layer class?  But I can accept List also..

File: Backend.Core/ItemsManagers/LayersManager.cs [hunk 2]
```diff
@@ -2,33 +2,65 @@
 using Backend.Models.RoomItems;
 using System.Collections.Generic;
 using System.Linq;
+using Serilog;
 namespace Backend.Core
 {
     internal class LayersManager : IRoomItemsManager
     {
-        private readonly Dictionary<string, Layer> _layers = new Dictionary<string, Layer>();
+        //private readonly Dictionary<string, Layer> _layers = new Dictionary<string, Layer>();
```
Comment (mikasso) [inline Backend.Core/ItemsManagers/LayersManager.cs:10]: remove the comment

File: Backend.Core/ItemsManagers/LayersManager.cs [hunk 3]
```diff
@@ -2,33 +2,65 @@
...
+        private List<Layer> _layers = new List<Layer>();
         public int Count => _layers.Count;
-        private List<Layer> LayerList { get => _layers.Values.ToList(); }
+        private int layerNameCount = 1;
```
Comment (mikasso) [inline Backend.Core/ItemsManagers/LayersManager.cs:13]: Please use property `Count `or change it to:
`private int NextLayerName get =>  $"Layer {Count+1}";`

For the future please follow the convention that private member variables are writen with _underscore ;)

File: Backend.Core/ItemsManagers/LayersManager.cs [hunk 4]
```diff
@@ -2,33 +2,65 @@
...
         }
 
-        public bool Delete(string Id) => _layers.Remove(Id);
+        public bool Delete(string Id)
+        {
+            bool result = _layers.Remove((Layer) Get(Id));
+            if (!result)
+            {
+                string layersStr = "";
+                for (int i = 0; i < _layers.Count; i++) layersStr += "\"" + _layers[i].Id + "\", ";
```
Comment (mikasso) [inline Backend.Core/ItemsManagers/LayersManager.cs:29]: use string interpolation $"{layseStr} text {_layers[i].Id}"

Comment (mikasso) [reply Backend.Core/ItemsManagers/LayersManager.cs:29]: Dont you miss brackets  { } for the loop?

File: Backend.Core/ItemsManagers/LayersManager.cs [hunk 5]
```diff
@@ -2,33 +2,65 @@
...
+                Log.Information($"Failed to delete layer: \"{Id}\" from layers: {layersStr}");
```
Comment (mikasso) [inline Backend.Core/ItemsManagers/LayersManager.cs:30]: Make it a warning, it indicates error in synchronization

**Review by mikasso — CHANGES_REQUESTED** [review]

File: Backend.Core/ItemsManagers/LayersManager.cs (hunk 1 above)
Comment (Piokor) [reply Backend.Core/ItemsManagers/LayersManager.cs:11]: I don't think it is worth to stick with Dictionary in this case. Layers' order is part of their logic, so List is more appropriate, there are also small amount of them, so Dict. indexing doesn't save significant amount of time

File: Backend.Core/ItemsManagers/LayersManager.cs (hunk 4 above)
Comment (Piokor) [reply Backend.Core/ItemsManagers/LayersManager.cs:29]: Yes

**Pull request merged**